src/rgw/rgw_http_client.cc

```diff
@@ -680,28 +680,50 @@ int RGWHTTPHeadersCollector::receive_header(void * const ptr, const size_t len)
 
   if (std::string_view::npos == sep_loc) {
     /* Wrongly formatted header? Just skip it. */
+    // Also!  Normal case!  Always call for crlf at end of headers.
     return 0;
   }
 
   header_name_t name(header_line.substr(0, sep_loc));
-  if (0 == relevant_headers.count(name)) {
+  bool interesting = relevant_headers.find(name) != relevant_headers.end();
```

Comment:
So the next person coming through this code doesn't have to figure out what "interesting" means, would you consider a juicy comment explaining what's happening here and below?